metrics-operator/controllers/common/providers/dynatrace/dynatrace.go

```diff
@@ -43,7 +43,12 @@ type DynatraceData struct {
 func (d *KeptnDynatraceProvider) EvaluateQuery(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) (string, []byte, error) {
 	baseURL := d.normalizeAPIURL(provider.Spec.TargetServer)
 	query := url.QueryEscape(metric.Spec.Query)
-	qURL := baseURL + "v2/metrics/query?metricSelector=" + query
+	var qURL string
+	if metric.Spec.Range != nil {
+		qURL = baseURL + "v2/metrics/query?metricSelector=" + query + "?writtenSince=" + metric.Spec.Range.Interval
```

Comment:
according to the DT API docs (https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/get-all-metrics), this should be:
```suggestion
		qURL = baseURL + "v2/metrics/query?metricSelector=" + query + "?writtenSince=now-" + metric.Spec.Range.Interval
```